tmt/convert.py

```diff
@@ -242,15 +243,32 @@ def read_datafile(
     # Test script
     search_result = re.search(regex_test, datafile, re.M)
     if search_result is None:
-        raise ConvertError("Makefile is missing the 'run' target.")
-    data['test'] = search_result.group(1)
-    if filename == 'metadata':
-        data['test'] = data['test'].split()[-1]
-    echo(style('test: ', fg='green') + data['test'])
+        if filename == 'metadata':
+            # entry_point property is optional. When absent 'make run' is used.
+            data['test'] = 'make run'
+        else:
+            raise ConvertError("Makefile is missing the 'run' target.")
+    else:
+        data['test'] = search_result.group(1)
+        echo(style('test: ', fg='green') + data['test'])
 
     # Detect framework
     try:
-        test_path = os.path.join(path, data["test"])
+        if data["test"].split()[0] != 'make':
+            test_path = os.path.join(path, data["test"].split()[-1])
+        else:
+            # As 'make' command was specified for test, ensure Makefile present.
+            makefile_path = os.path.join(path, 'Makefile')
+            try:
+                with open(makefile_path, encoding='utf-8') as makefile_file:
+                    makefile = makefile_file.read()
+                    search_result = \
+                        re.search(makefile_regex_test, makefile, re.M)
+            except IOError:
+                raise ConvertError("Makefile is missing.")
+            # Retrieve the path to the test file from the Makefile
+            test_path =  os.path.join(path, search_result.group(1).split()[-1])
+        # Read the test file and determine the framework used.
         with open(test_path, encoding="utf-8") as test_file:
```

Comment:
test_file doesn't need to be file, it is allowed to be a command  as well. tmt should attempt to determine framework but should cope with file missing.

Reply:
I've updated the logic in the code. The value for entry_point will be checked. If neither a 'make' command nor a bash script are found within then it it assumed to be a command and so the framework will be set to 'shell' without the need to check a file. That way tmt will execute the command and accept the return code as pass/fail or the command.